odoo/modules/registry.py

```diff
@@ -345,12 +345,30 @@ def field_computed(self):
                     computed[field] = group = groups[field.compute]
                     group.append(field)
             for fields in groups.values():
+                if len(fields) < 2:
+                    continue
                 if len({field.compute_sudo for field in fields}) > 1:
-                    _logger.warning("%s: inconsistent 'compute_sudo' for computed fields: %s",
-                                    model_name, ", ".join(field.name for field in fields))
+                    fnames = ", ".join(field.name for field in fields)
+                    warnings.warn(
+                        f"{model_name}: inconsistent 'compute_sudo' for computed fields {fnames}. "
+                        f"Either set 'compute_sudo' to the same value on all those fields, or "
+                        f"use distinct compute methods for sudoed and non-sudoed fields."
+                    )
```

Comment:
Those warnings indeed show up each once per compute method (once for all the fields of that compute method).
And it's made there because that's when you determine which fields are computed by the same method (which is lazily determined).
Do you have an alternative proposal?